### PR TITLE
Add navigation links documentation

### PR DESCRIPTION
**Pull request by oscarwyatt**
Configuring the navigation links out is fairly complicated and it is not immediately obvious
what it does or how it works. This adds documentation covering the taxonomy_navigation_links_out.yml file and how to edit it

---

Visual regression results:
https://government-frontend-pr-1073.surge.sh/gallery.html

Component guide for this PR:
https://government-frontend-pr-1073.herokuapp.com/component-guide

Trello card:
https://trello.com/c/UnIYjmGr/143-start-documenting-work-done-this-quarter-on-the-wiki

### CODE REVIEW

**Comment by vanitabarrett** [comment]
Can we close https://github.com/alphagov/government-frontend/pull/1068 and replace it with this PR?

**Comment by vanitabarrett** [comment]
@oscarwyatt I've merged https://github.com/alphagov/government-frontend/pull/1068, so you should be able to rebase this PR so it only shows the relevant commit

File: README.md [hunk 1]
```diff
@@ -48,6 +48,8 @@ Components specific to government-frontend are [within the application](https://
 
 When [running government-frontend](#running-the-application) locally the component guide is available at: http://government-frontend.dev.gov.uk/component-guide
 
+Configuration for the navigation links is covered [separately](docs/navigation-links.md)
```
Comment (vanitabarrett) [inline README.md]: I would try and make it clear here that these are the navigation links displayed as part of the content pages navigation AB test. We can update the docs later if we decide to roll this out permanently.

Comment (oscarwyatt) [reply README.md]: Good shout 👍

**Pull request merged**